harness/determined/gpu.py

```diff
@@ -1,16 +1,60 @@
+import csv
+import logging
+import subprocess
 import sys
-from typing import List, Optional, Tuple
+from typing import List, NamedTuple, Optional, Tuple
 
 from determined_common import check
 
+gpu_fields = [
+    "index",
+    "uuid",
+    "utilization.gpu",
+    "memory.used",
+    "memory.total",
+]
 
-def get_gpu_ids_and_uuids() -> Tuple[List[str], List[str]]:
+
+class GPU(NamedTuple):
+    id: int
+    uuid: str
+    load: float
+    memoryUtil: float
+
+
+def get_gpus() -> List[GPU]:
     try:
-        import GPUtil
-    except ModuleNotFoundError:
-        return [], []
+        proc = subprocess.Popen(
+            ["nvidia-smi", "--query-gpu=" + ",".join(gpu_fields), "--format=csv,noheader,nounits"],
+            stdout=subprocess.PIPE,
+            universal_newlines=True,
+        )
+    except (FileNotFoundError, subprocess.CalledProcessError):
+        return []
```

Comment:
nonblocking: could be good to log a warning and more information when `nvidia-smi` encounters an error vs. is not found. The not found case is expected behavior with CPU trials, but found and returned an error is an unexpected case a user might want to debug further.